pkg/controllers/edgeconnect/controller.go

```diff
@@ -667,6 +661,32 @@ func (controller *Controller) createOrUpdateEdgeConnectDeployment(ctx context.Co
 	return nil
 }
 
-func edgeConnectClientSecretName(edgeConnectName string) string {
-	return edgeConnectName + "-client"
+func (controller *Controller) createOrUpdateEdgeConnectConfigSecret(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) error {
+	configFile, err := secret.PrepareConfigFile(ctx, edgeConnect, controller.apiReader)
+	if err != nil {
+		return err
+	}
+
+	secretData := make(map[string][]byte)
+	secretData[consts.EdgeConnectConfigName] = configFile
+
+	secretConfig, err := k8ssecret.Create(controller.scheme, edgeConnect,
+		k8ssecret.NewNameModifier(consts.EdgeConnectConfigVolumeMountName),
```

Comment:
the name of the secret shouldn't be a const
You can have multiple EdgeConnects, so each configuration should have its unique name derived from the name of the EdgeConnect that owns it.

Things to look out for:
- Be careful with the char limit of resource names
- Just "trimming" the name to fit in the limit might lead to conflicts
- AFAIK we have a validation warning for the DynaKube's name's length for the same reason.

Reply:
yes that works but don't forget

> Be careful with the char limit of resource names
Just "trimming" the name to fit in the limit might lead to conflicts
AFAIK we have a validation warning for the DynaKube's name's length for the same reason.

Reply:
okey let me try out it.

Reply:
fixed in 095bc819